app/Helpers/PaymentService.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace App\Helpers;
+
+use App\Models\CustomerCart;
+use App\Models\HistoryTransaction;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Log;
+use Ixudra\Curl\Facades\Curl;
+
+class PaymentService
+{
+
+
+    public static function CheckOut($product)
+    {
+
+        try {
+            $qty = 0;
+            $items = [];
+            $cartIds = [];
+            foreach ($product['products'] as $prod) {
+                $qty += $prod['qty'];
+                $cartIds[] = $prod['id'];
+                $items[] =      [
+                    'name' => $prod['name'],
+                    'quantity' => $prod['qty'],
+                    'description' => 'Description of the test product',
+                    'amount' => [
+                        'value' => $prod['price'],
+                        'details' => [
+                            'subtotal' => $prod['price'] * $prod['qty'],
+                            'discount' => 0,
+                            'serviceCharge' => 0,
+                            'shippingFee' => 0,
+                            'tax' => 0
+                        ]
+                    ],
+                    'totalAmount' => [
+                        'currency' => 'PHP',
+                        'value' => $prod['price'] * $prod['qty'],
+                        'details' => [
+                            'subtotal' => $prod['price'] * $prod['qty']
+                        ]
+                    ]
+                ];
+            }
```

Comment:
Kindly create a different method that will generate the checkout payload. 

As a rule of thumb, let's write our functions with only (1) purpose in mind. Right now, CheckOut is bogged down by product summary, payload generation, API consumption, and HistoryTransaction creation. It is best practice to separate each concern to their own functions.